### ISSUE TITLE
Extract XML as pugi_document

### ISSUE THREAD

**Issue by mr-j0nes** [zeux/pugixml#154]
Hi 
I am looking for a way to extract part of an XML document into another pugi_document. Is it possible ?
Say I have following XML:
``` xml
<main_config>
    <debug>1</debug>
    <enemies>
        <ninja>
            <powers>
                <blahblah1/>
            </powers>
        </ninja>
        <monster>
            <powers>
                <blahblah2/>
            </powers>
        </monster>
    </enemies>
</main_config>
```
Then I have two classes `Ninja` and `Monster` and each of them has a member `pugi::xml_document` which should be populated with the corresponding branch of `powers`. Any way of doing so ?

I checked the `xml_node::root()` method but that extracts a `xml_node` instead of a `xml_document`.
Also I checked the `xml_document::reset()` method but in turn that accepts only `xml_document`, no `xml_node`

**Comment by zeux** [comment]
Are you looking for something like this? http://pugixml.org/docs/manual.html#modify.clone

**Comment by mr-j0nes** [comment]
Hello there, thanks for the answer. Yes that's what I needed. I found it on an already closed issue #153.
